### DISCUSSION TITLE
forceDeleting and restoring events not available for observers

### DISCUSSION THREAD

**Discussion by ekmungai** [laravel/framework#50447] [Q&A]
### Laravel Version

10.47.0

### PHP Version

8.1.4

### Description

The two events throw a `BadMethodCallException: Call to undefined method App\Models\User::forceDeleting()`  and `BadMethodCallException: Call to undefined method App\Models\User::restoring()` exception when attempting to boot the model

### Steps To Reproduce

Create a class with closure listeners for the above events 

```php
class User extends Model
{
    /**
     * The "booted" method of the model.
     */
    protected static function booted(): void
    {
        static::restoring(function (User $user) {
            // do something
        });
    }
    
    ...
}
```
Then try to initialise the model

```shell
PS C:\projects\example-app> php artisan tinker
Psy Shell v0.12.0 (PHP 8.1.4 — cli) by Justin Hileman
> $user = new User();
[!] Aliasing 'User' to 'App\Models\User' for this Tinker session.

   BadMethodCallException  Call to undefined method App\Models\User::restoring().

>
```

**Comment by staudenmeir** [comment]
Hi @ekmungai,
You can only listen to these events if your model is actually [soft-deletable](https://laravel.com/docs/eloquent#soft-deleting). You need to add the `SoftDeletes` trait:

```php
class User extends Model
{
    use SoftDeletes;
 
    // ...
}
```
